searchlib/src/tests/queryeval/nearest_neighbor/nearest_neighbor_test.cpp

```diff
@@ -190,4 +191,70 @@ TEST("require that NearestNeighborIterator sets expected rawscore") {
     TEST_DO(verify_iterator_sets_expected_rawscore(denseSpecFloat, denseSpecDouble));
 }
 
+TEST("require that NnsIndexIterator works as expected") {
+    std::vector<NnsIndexIterator::Hit> hits{{2,4.0}, {3,9.0}, {5,1.0}, {8,16.0}, {9,36.0}};
+    auto md = MatchData::makeTestInstance(2, 2);
+    auto &tfmd = *(md->resolveTermField(0));
+    auto search = NnsIndexIterator::create(true, tfmd, hits);
+    uint32_t docid = 1;
+    search->initFullRange();
+    bool match = search->seek(docid);
+    EXPECT_FALSE(match);
+    EXPECT_FALSE(search->isAtEnd());
+    EXPECT_EQUAL(2u, search->getDocId());
+    docid = 2;
+    match = search->seek(docid);
```

Comment:
To make the test easier to read, consider creating some helper functions: one that seeks and expects a match with associated raw score. The other that seeks but doesn't expect a match, but checks current doc id on the iterator. The last that seeks and expects to be at the end.